src/transformStateWithClones.js

```diff
@@ -1,13 +1,48 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object} stateClone
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateClone = {};
+
+  Object.assign(stateClone, state);
+
+  const resultArray = [];
+
+  for (const action of actions) {
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        resultArray.push(Object.assign({}, stateClone));
```

Comment:
don't push object to resultArray in each if you can do it once at the end of the loop